bn_mp_factors_init.c

```diff
@@ -0,0 +1,32 @@
+#include "tommath_private.h"
+#ifdef BN_MP_FACTORS_INIT_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis
+ *
+ * LibTomMath is a library that provides multiple-precision
+ * integer arithmetic as well as number theoretic functionality.
+ *
+ * The library was designed directly after the MPI library by
+ * Michael Fromberger but has been written from scratch with
+ * additional optimizations in place.
+ *
+ * SPDX-License-Identifier: Unlicense
+ */
+
+/* Init factor-array (an array of mp_int's) */
+#ifdef LTM_USE_EXTRA_FUNCTIONS
+int mp_factors_init(mp_factors *f)
+{
+   f->factors = OPT_CAST(mp_int) XMALLOC(sizeof(mp_int) * (size_t)(LTM_TRIAL_GROWTH));
```

Comment:
this should be cleared, shouldn't it? so it could be a `XCALLOC()`!?

which brings me directly into... why aren't `mp_init()` and `mp_init_size()` using `XCALLOC()`?

Reply:
> this should be cleared, shouldn't it?

It is more or less just the backbone and gets overwritten. On the other side: better save than sorry and it doesn't cost much, so I'll clear it.

> which brings me directly into... why aren't mp_init() and mp_init_size() using XCALLOC()?

It might have had a reason once but now? Yes. would support a change to XCALLOC here.

Reply:
Lunchbreak wasn't shortened, so I was able to rebase before I *really* need to run and that reminded me that  `XCALLOC` isn't in LTM anymore. Do you want to put it back in?